### PR TITLE
Add raydium CLMM oracle fallbacks

### PR DESCRIPTION
**Pull request by Lou-Kamades**

### CODE REVIEW

File: programs/mango-v4/src/state/oracle.rs [hunk 1]
```diff
@@ -404,56 +408,32 @@ fn oracle_state_unchecked_inner<T: KeyedAccountReader>(
...
-                last_update_slot: quote_state.last_update_slot,
-                deviation: quote_state.deviation,
+                last_update_slot: quote_oracle_state.last_update_slot,
+                deviation: quote_oracle_state.deviation,
                 oracle_type: OracleType::OrcaCLMM,
             }
         }
+        OracleType::RaydiumCLMM => {
+            let whirlpool = load_raydium_pool_state(oracle_info)?;
+            let clmm_price = whirlpool.get_clmm_price();
```
Comment (microwavedcola1) [inline programs/mango-v4/src/state/oracle.rs:424]: is this the twap price?

Comment (Lou-Kamades) [reply programs/mango-v4/src/state/oracle.rs:424]: No, it's just the raw price on the AMM

File: programs/mango-v4/src/state/oracle.rs [hunk 2]
```diff
@@ -12,9 +12,9 @@ use switchboard_v2::AggregatorAccountData;
 use crate::accounts_zerocopy::*;
 
 use crate::error::*;
-use crate::state::load_whirlpool_state;
+use crate::state::load_orca_pool_state;
 
-use super::orca_mainnet_whirlpool;
+use super::{load_raydium_pool_state, orca_mainnet_whirlpool, raydium_mainnet};
```
Comment (ckamm) [inline programs/mango-v4/src/state/oracle.rs:17]: importing the orca function and raydium function from different modules looks weird

File: programs/mango-v4/src/state/amm_cpi.rs [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    );
+    require!(
+        acc_info.owner() == &raydium_mainnet::ID,
+        MangoError::InvalidCLMMOracle
+    );
+
+    let price_bytes: &[u8; 16] = &data[253..269].try_into().unwrap();
+    let sqrt_price = u128::from_le_bytes(*price_bytes);
+    let a: &[u8; 32] = &(&data[73..105]).try_into().unwrap();
+    let b: &[u8; 32] = &(&data[105..137]).try_into().unwrap();
```
Comment (ckamm) [inline programs/mango-v4/src/state/amm_cpi.rs:159]: todo: verify offsets

**Review by ckamm — COMMENTED** [review]
I like the tests!

**Pull request merged**